velox/connectors/hive/HiveConfig.cpp

```diff
@@ -124,11 +124,21 @@ bool HiveConfig::ignoreMissingFiles(const config::ConfigBase* session) const {
 }
 
 int64_t HiveConfig::maxCoalescedBytes() const {
-  return config_->get<int64_t>(kMaxCoalescedBytes, 128 << 20);
+  return config_->get<int64_t>(kMaxCoalescedBytes, 128 << 20); // 128MB
 }
 
-int32_t HiveConfig::maxCoalescedDistanceBytes() const {
-  return config_->get<int32_t>(kMaxCoalescedDistanceBytes, 512 << 10);
+int32_t HiveConfig::maxCoalescedDistanceBytes(
+    const config::ConfigBase* session) const {
+  const auto distance = config::toCapacity(
+      session->get<std::string>(
+          kOrcMaxMergeDistanceSession,
+          config_->get<std::string>(kMaxCoalescedDistance, "512kB")),
+      config::CapacityUnit::BYTE);
+  // Given distance should not be even close to 2GB (max int32), but just in
+  // case add a check to avoid weird overflow issues.
+  return distance <= std::numeric_limits<int32_t>::max()
+      ? distance
+      : std::numeric_limits<int32_t>::max();
```

Comment:
Maybe just throw error in this case?  People should be aware the value they put in here is not working as they would expect